test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -8,13 +8,151 @@ var chai = require('chai')
+          });
+        });
+        
+      });
+
+      it('produces an error when missing expected parameters', function () {
+        var self = this;
+        var body = 'return 1;';
+        var options = {};
+        
+        //requires functionName  
+        expect( function () {
+          self.queryInterface.createFunction(null, [{name:'test'}], 'integer', 'plpgsql', body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+      
+        //requires Parameters array
+        expect( function () {
+          self.queryInterface.createFunction('create_job',null, 'integer', 'plpgsql', body, options);
+        }).to.throw(/function parameters array required/);
+        
+        //requires returnType
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], null, 'plpgsql', body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+      
+        //requires type in parameter array
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{name:'test'}], 'integer', 'plpgsql', body, options);
+        }).to.throw(/function or trigger used with a parameter without any type/);
+        
+        //requires language
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', null, body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+
+        //requires body
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', null, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+
+      });
+    });
+
+    describe('dropFunction',function () {
+      beforeEach( function () {
+
+        var body = 'return test;';
+        var options = {};
+
+        //make sure we have a droptest function in place.
+        return this.queryInterface.createFunction('droptest', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', body, options)
+        //suppress errors.. this could fail if the function is already there.. thats ok. 
+        .catch( function (err) { });
+      });
+
+      it('can drop a function', function () {
+        var self = this;
+         
+        //call dropFunction
+        return self.queryInterface.dropFunction('droptest',[{type:'varchar',name:'test'}])
+        //validate
+        .then( function () {
+          return self.sequelize.query('select droptest(\'test\');', { type: self.sequelize.QueryTypes.SELECT })
+          .then( function () {
+            throw new Error('function failed to drop');
```

Comment:
Here is my understandings so far

- `dropFunction` removes the `droptest`
- Try to call the function `droptest` (note it has been dropped in last statement), So we are calling a dropped function. This statement should be rejected
- Add `catch` handler  or expect `error` to be thrown

But you are expecting `.then()` in Step 3, then throw error yourself to catch it again why ? If `dropFunction` really fails to remove the function in Step 1, mocha should catch the failed promise.

You should do something like

```js
return self.queryInterface
.dropFunction('droptest',[{type:'varchar',name:'test'}])
.then(function () {
  return expect(self.sequelize.query('select droptest(\'test\');', {
    type: self.sequelize.QueryTypes.SELECT
  }))
  .to.eventually.rejectedWith(/function droptest.* does not exist/);
});
```